src/smolagents/local_python_executor.py

```diff
@@ -608,16 +616,29 @@ def evaluate_call(
     authorized_imports: List[str],
 ) -> Any:
     if not (
-        isinstance(call.func, ast.Attribute) or isinstance(call.func, ast.Name) or isinstance(call.func, ast.Subscript)
+        callable(call.func)
+        or isinstance(call.func, ast.Call)
```

Comment:
You can combine them as `isinstance(call.func, (ast.Attribute, ast.Name, ast.Call, ast.Lambda, ...))`.

Reply:
True. I just followed what the guys did before. Maybe they don't like this approach 🤷  
Honestly I would have replaced that with a match operator instead.

Reply:
> Honestly I would have replaced that with a match operator instead.

Agreed with this.